src/Codec/CBOR/Cuddle/CBOR/Gen.hs

```diff
@@ -77,8 +81,34 @@ data GenState g = GenState
   }
   deriving (Generic)
 
+instance RandomGen g => RandomGen (GenState g) where
+  genWord8 = withRandomSeed genWord8
+  genWord16 = withRandomSeed genWord16
+  genWord32 = withRandomSeed genWord32
+  genWord64 = withRandomSeed genWord64
+#if MIN_VERSION_random(1,3,0)
+  split = error "Split was deprecated and should no longer used. Use `splitGen` instead"
+#else
```

Comment:
```suggestion
#if !MIN_VERSION_random(1,3,0)
```

Reply:
It's not so simple. Your suggestion will require `SplitGen g =>` superclass constraint, but only for `random-1.3`, since default implementation falls back on `splitGen`.

The correct way to approach this would be to use the same implementation as in `random-1.2`:
```haskell
  split = splitGenStateWith split
```

However, this would require `{-# OPTIONS_GHC -Wno-deprecations #-}`, since it will result in a warning due to usage of `split` being deprecated. Because of this I am a bit on the fence here.

Reply:
I think it will be safer to avoid using `error`, so disabling of this warning is warranted IMHO.